src/ui/reducers/app.ts

```diff
@@ -252,6 +252,30 @@ export const getPointsForHoveredLineNumber = (state: UIState) => {
 const NO_EVENTS: MouseEvent[] = [];
 export const getEventsForType = (state: UIState, type: string) =>
   state.app.events[type] || NO_EVENTS;
+
+// This will return an object with each category (mouse/keyboard/navigation)
+// as the key and a boolean value indicating whether it's still loading
+export const getEventCategoriesLoading = (state: UIState) => {
```

Comment:
This should probably be memoized with reselect

Reply:
There's not much to optimize here — as far as I've seen, this only gets computed <3 times. As for complexity, the category (e.g. `mouse`) matching to the event kinds (`mousedown`)  doesn't get bad at all, since there's a limited amount of permutations (max `3 * 7 = 21`) every time the selector runs..